modules/javafx.graphics/src/main/native-glass/mac/GlassWindow.m

```diff
@@ -1037,32 +1037,45 @@ static jlong _createWindowCommonDo(JNIEnv *env, jobject jWindow, jlong jOwnerPtr
     GLASS_POOL_ENTER;
     {
         GlassWindow *window = getGlassWindow(env, jPtr);
+
         window->suppressWindowResizeEvent = YES;
+        window->suppressWindowMoveEvent = YES;
 
-        if ((maximize == JNI_TRUE) && (isZoomed == JNI_FALSE))
-        {
-            window->preZoomedRect = [window->nsWindow frame];
+        NSRect oldFrame = window->nsWindow.frame;
+        int eventType = com_sun_glass_events_WindowEvent_RESTORE;
 
-            if ([window->nsWindow styleMask] != NSWindowStyleMaskBorderless)
-            {
+        if ((maximize == JNI_TRUE) && (isZoomed == JNI_FALSE)) {
```

Comment:
changing this code to K&R style messes up the diff alignment...  could we retain the original style?